pkg/controllers/dynakube/extension/eec/reconciler.go

```diff
@@ -33,6 +33,8 @@ func NewReconciler(clt client.Client, apiReader client.Reader, dk *dynakube.Dyna
 }
 
 func (r *reconciler) Reconcile(ctx context.Context) error {
+	defer r.deleteLegacyStatefulset(ctx)
```

Comment:
do we always want to spam the API with deletes? (even when no extensions is/was configured?)

why not have the `defer` inside/after the `if`? (you would need 2 `defer` which is ugly, but the expected "business" logic itself is ugly, so you can't really help it 😅 )

Reply:
sure, our cleanup isn't reliable anyways